src/main/scala/tech/cryptonomic/conseil/tezos/TezosNodeOperator.scala

```diff
@@ -32,61 +32,81 @@ object TezosNodeOperator {
     */
   final case class OperationResult(results: AppliedOperation, operationGroupID: String)
 
+  val isGenesis = (data: BlockData) => data.header.level == 0
+
+  /** Helper to decode json and convert to a possibly failing future result
+    * This is not running any async operation
+    */
+  def decodeToFuture[A: io.circe.Decoder](json: String) = {
+    import io.circe.parser.decode
+
+    decode[A](json) match {
+      case Left(error) => Future.failed(error)
```

Comment:
We should log the failing JSON string to console. Maybe you can do this in the subsequent PR?

Reply:
I'm actually considering how to "percolate" all relevant failure information to the top level, even from multiple threads.
The idea is to have a way to get detailed information on each step failure from the bottom up, traced with some id, such that we can rebuild some narrative of the event through all the layers.